pilot/pkg/config/file/store.go

```diff
@@ -433,17 +437,69 @@ func (s *KubeSource) parseChunk(r *collection.Schemas, name string, lineNum int,
 	}
 
 	pos := kube2.Position{Filename: name, Line: lineNum}
-	c, err := ToConfig(objMeta, schema, &pos, fieldMap)
+	c, err := ToConfig(objMeta, schema, &pos, fieldMap, false)
 	if err != nil {
-		return resources, err
+		return nil, err
 	}
-	return []kubeResource{
+	cfgs := []kubeResource{
 		{
 			schema: schema,
 			sha:    sha256.Sum256(yamlChunk),
 			config: c,
 		},
-	}, nil
+	}
+	_, found = r.FindByGroupVersionAliasesKind(gvk.Pod)
+	if isWorkload(c.GroupVersionKind) && found {
+		pod := generateAnalyzePodFromWorkloadTemplate(c)
+		if pod == nil {
+			return cfgs, nil
+		}
+		pc, err := ToConfig(pod, collections.Pod, &pos, nil, true)
+		if err != nil {
+			return cfgs, err
+		}
+		cfgs = append(cfgs, kubeResource{
+			schema: collections.Pod,
+			sha:    sha256.Sum256(yamlChunk),
+			config: pc,
+		})
+	}
+	return cfgs, nil
+}
+
+var WorkloadKinds = map[config.GroupVersionKind]bool{
```

Comment:
@howardjohn if we expand this list to inclued Daemonsets, statefulsets, and replicasets, would that alleviate your concern?

Reply:
My comment was on an old PR where we only read Pods directly. Reading more than deployment would help but this is already addressing my primary concern I think